lib/pipeline.js

```diff
@@ -379,14 +379,16 @@ class PipelineModel extends BaseModel {
      * @return  {Promise}
      */
     addWebhook(webhookUrl) {
-        return this.token.then(token =>
-            this.scm.addWebhook({
-                scmUri: this.scmUri,
-                scmContext: this.scmContext,
-                token,
-                webhookUrl
-            })
-        );
+        return this.getFirstRepoAdmin()
```

Comment:
do we have checks before reaching here that the user has atleast write permissions ? If not we might end up allowing anyone to add webhook.

Reply:
Thanks. That is checked at [this code](https://github.com/screwdriver-cd/screwdriver/blob/ce6822c9c9603f1c6075365ff02eea9a33824d16/plugins/pipelines/syncWebhooks.js#L44).